test/status_file.c

```diff
@@ -0,0 +1,366 @@
+
+	g_assert_nonnull(r_context()->config->statusfile_path);
+
+	/* Set status for all slots */
+	g_hash_table_iter_init(&iter, slots);
+	while (g_hash_table_iter_next(&iter, NULL, (gpointer*) &slot)) {
+		if (slot->status)
+			r_slot_free_status(slot->status);
+
+		g_debug("Set default status for slot %s.", slot->name);
+		slot->status = g_new0(RaucSlotStatus, 1);
+		slot->status->status = g_strdup("ok");
+		slot->status->checksum.type = G_CHECKSUM_SHA256;
+		slot->status->checksum.digest = g_strdup("dc626520dcd53a22f727af3ee42c770e56c97a64fe3adb063799d8ab032fe551");
+	}
+
+	/* Save status for all slots */
+	g_hash_table_iter_init(&iter, slots);
+	while (g_hash_table_iter_next(&iter, NULL, (gpointer*) &slot)) {
+		res = r_slot_status_save(slot, &ierror);
+		g_assert_no_error(ierror);
+		g_assert_true(res);
+	}
+
+	/* Clear status for all slots */
+	g_hash_table_iter_init(&iter, slots);
+	while (g_hash_table_iter_next(&iter, NULL, (gpointer*) &slot)) {
+		if (slot->status)
+			r_slot_free_status(slot->status);
+
+		slot->status = NULL;
+	}
+
+	/* Check status for all slots */
+	g_hash_table_iter_init(&iter, slots);
+	while (g_hash_table_iter_next(&iter, NULL, (gpointer*) &slot)) {
+		r_slot_status_load(slot);
+		g_assert_nonnull(slot->status);
+		g_assert_cmpstr(slot->status->status, ==, "ok");
+		g_assert_cmpint(slot->status->checksum.type, ==, G_CHECKSUM_SHA256);
+		g_assert_cmpstr(slot->status->checksum.digest, ==,
+				"dc626520dcd53a22f727af3ee42c770e56c97a64fe3adb063799d8ab032fe551");
+	}
+}
+
+/* Loads system status from a file containing only system status information */
+static void status_file_test_load_system_status(ConfigFileFixture *fixture,
+		gconstpointer user_data)
+{
+	g_autofree gchar* pathname = NULL;
+	GError *ierror = NULL;
+	gboolean res;
+	g_autoptr(RSystemStatus) system_status = NULL;
+
+	const gchar *cfg_file = "\
+[system]\n\
+boot-id=924ebd2e-c85f-4c48-b92d-cd1b378d9994\n\
+";
+	pathname = write_tmp_file(fixture->tmpdir, "system_only.conf", cfg_file, NULL);
+	g_assert_nonnull(pathname);
+
+	system_status = g_new0(RSystemStatus, 1);
+	g_assert_nonnull(system_status);
+
+	/* assert error-free loading */
+	res = r_system_status_load(pathname, system_status, &ierror);
+	g_assert_no_error(ierror);
+	g_assert_true(res);
+
+	/* assert loaded data */
+	g_assert_cmpstr(system_status->boot_id, ==, "924ebd2e-c85f-4c48-b92d-cd1b378d9994");
+}
+
+/* Loads a broken system status */
+static void status_file_test_load_broken(ConfigFileFixture *fixture,
+		gconstpointer user_data)
+{
+	gchar* pathname = NULL;
+	GError *ierror = NULL;
+	gboolean res;
+	g_autoptr(RSystemStatus) system_status = NULL;
+
+	const gchar *cfg_file = "\
+[system]\n\
+boot-id=924ebd2e-c85f-4c48-b92d-cd1b378d9994\n\
+\n\
+[broken\n\
+";
+	pathname = write_tmp_file(fixture->tmpdir, "broken_config.conf", cfg_file, NULL);
+	g_assert_nonnull(pathname);
+
+	system_status = g_new0(RSystemStatus, 1);
+	g_assert_nonnull(system_status);
+
+	/* assert error-free loading */
```

Comment:
This comment doesn't match the code.

Reply:
May contain traces of copy-paste :smirk: